ou_dedetai/system.py

```diff
@@ -232,6 +235,67 @@ def get_dialog():
         config.DIALOG = 'tk'
 
 
+def get_architecture():
+    machine = platform.machine().lower()
+    bits = struct.calcsize("P") * 8
+
+    if "x86_64" in machine or "amd64" in machine:
+        architecture = "x86_64"
+    elif "i386" in machine or "i686" in machine:
+        architecture = "x86_32"
+    elif "arm" in machine or "aarch64" in machine:
+        if bits == 64:
+            architecture = "ARM64"
+        else:
+            architecture = "ARM32"
+    elif "riscv" in machine or "riscv64" in machine:
+        if bits == 64:
+            architecture = "RISC-V 64"
+        else:
+            architecture = "RISC-V 32"
+    else:
+        architecture = "Unknown"
+
+    return architecture, bits
```

Comment:
something I just thought of now, is there a standard for this enum? Some spec or other source? seems like this string varies wildly depending on the software. It'd be nice to refer to some other sheet for this (instead of adding one more combination to the software world)

For example we could use FreeBSD's codes, they have a nice table here: https://www.freebsd.org/platforms/#_supported_platforms